ui/tests/integration/components/search-select-test.js

```diff
@@ -0,0 +1,164 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { create } from 'ember-cli-page-object';
+import { typeInSearch, clickTrigger } from 'ember-power-select/test-support/helpers';
+import Service from '@ember/service';
+import { render } from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+import searchSelect from '../../pages/components/search-select';
+
+const component = create(searchSelect);
+
+const storeService = Service.extend({
+  query(modelType) {
+    return new Promise((resolve, reject) => {
+      switch (modelType) {
+        case 'policy/acl':
```

Comment:
We also have Pretender which can make this a little nicer - I don't think you need to move to that - but you can mock at the Ajax level that way, and specify the response for each endpoint / verb.